.github/workflows/build.yml

```
@@ -201,3 +201,65 @@ jobs:
           repo_token: ${{ secrets.GITHUB_TOKEN }}
           tag: ${{ env.ONNXRUNTIME_VERSION }} # ==> github.event.release.tag_name
           file: ${{ env.RELEASE_NAME }}.tgz
+  
+  build-xcframework:
+    needs: build-onnxruntime
+    runs-on: macos-12
+    steps:
+      - name: Generate RELEASE_NAME and ONNXRUNTIME_BASENAME
+        run: |
+          echo "RELEASE_NAME=onnxruntime-xcframework-${{ env.ONNXRUNTIME_VERSION }}" >> $GITHUB_ENV
+          echo "ONNXRUNTIME_BASENAME=libonnxruntime.${{ env.ONNXRUNTIME_VERSION }}.dylib" >> "$GITHUB_ENV"
+
+      - uses: actions/download-artifact@v2
+        with:
+          name: onnxruntime-ios-arm64
+          path: artifact/onnxruntime-aarch64-apple-ios
+
+      - uses: actions/download-artifact@v2
+        with:
+          name: onnxruntime-ios-sim-arm64
+          path: artifact/onnxruntime-aarch64-apple-ios-sim
+
+      - uses: actions/download-artifact@v2
+        with:
+          name: onnxruntime-ios-sim-x86_64
+          path: artifact/onnxruntime-x86_64-apple-ios
+
+      - name: Remove no version notation dylib
+        shell: bash
+        run: |
+          rm -f artifact/onnxruntime-x86_64-apple-ios/lib/*onnxruntime.dylib
+          rm -f artifact/onnxruntime-aarch64-apple-ios-sim/lib/*onnxruntime.dylib
+          rm -f artifact/onnxruntime-aarch64-apple-ios/lib/*onnxruntime.dylib
+
+      - name: Create fat binary
+        shell: bash
+        run: |
+          mkdir -p "artifact/onnxruntime-sim"
+          lipo -create "artifact/onnxruntime-x86_64-apple-ios/lib/${{ env.ONNXRUNTIME_BASENAME }}" "artifact/onnxruntime-aarch64-apple-ios-sim/lib/${{ env.ONNXRUNTIME_BASENAME }}" -output "artifact/onnxruntime-sim/${{ env.ONNXRUNTIME_BASENAME }}"
+
+      - name: Create XCFramework
+        shell: bash
+        run: |
+          mkdir -p "artifact/${{ env.ONNXRUNTIME_BASENAME }}"
+          xcodebuild -create-xcframework \
+            -library "artifact/onnxruntime-sim/${{ env.ONNXRUNTIME_BASENAME }}" \
+            -library "artifact/onnxruntime-aarch64-apple-ios/lib/${{ env.ONNXRUNTIME_BASENAME }}" \
+            -output "artifact/${{ env.ONNXRUNTIME_BASENAME }}/onnxruntime.xcframework"
+
+      - name: Archive artifact
+        shell: bash
+        run: |
+          cd artifact/${{ env.ONNXRUNTIME_BASENAME }}
+          7z a "../../${{ env.RELEASE_NAME }}.zip" "onnxruntime.xcframework"
+
+      - name: Upload to Release
+        if: env.RELEASE == 'true'
+        uses: softprops/action-gh-release@v1
+        with:
+          prerelease: true
+          tag_name: ${{ env.VERSION }}

```

`VERSION` 変数は存在しないので `ONNXRUNTIME_VERSION` を使う必要がありそうです。

```suggestion
          tag_name: ${{ env.ONNXRUNTIME_VERSION }}
```

また、`build-onnxruntime` ジョブの方では release へのファイルのアップロードに `softprops/action-gh-release@v1` ではなく `svenstaro/upload-release-action@v2` を使っていて、ちょっと統一感がないのが気になりました（問題はなさそう？）。

確認したところ、コアとエディタのリポジトリでは `softprops/action-gh-release@v1`, エンジンでは `ncipollo/release-action@v1` を使っているようで、何の action を使うかは VOICEVOX プロジェクトとしては特に方針はなかったのかな、という感じでした。この辺り、どのようにしましょうか……？ @Hiroshiba

---

> `VERSION` 変数は存在しないので `ONNXRUNTIME_VERSION` を使う必要がありそうです。

ありがとうございます。voicevox_coreリポジトリから移植したままになっていました。

> 特に方針はないですが、少なくとも同じリポジトリでは統一されていた方が無難に思いました。

わかりました。それでは、このリポジトリで以前から使われていた`svenstaro/upload-release-action@v2`へ変更します。

いずれも dc8d51d で対応しました。